apps/prairielearn/elements/pl-drawing/mechanicsObjects.js

```diff
@@ -3046,10 +3046,12 @@ mechanicsObjects.byType['pl-paired-vector'] = class extends PLDrawingBaseElement
       'widthErrorBox',
       'heightErrorBox',
     ];
-    for (i = 0; i < varyingAttributes.length; i++) {
-      if (typeof options[varyingAttributes[i].concat('1')] != typeof options['abcnonsense']) {
-        options1[varyingAttributes[i]] = options[varyingAttributes[i].concat('1')];
-        options2[varyingAttributes[i]] = options[varyingAttributes[i].concat('2')];
+    for (const varyingAttribute of varyingAttributes) {
+      if (varyingAttribute.concat('1') in options) {
+        options1[varyingAttribute] = options[varyingAttribute.concat('1')];
+      }
+      if (varyingAttribute.concat('2') in options) {
+        options2[varyingAttribute] = options[varyingAttribute.concat('2')];
```

Comment:
This looks like it may have been a bug, though it probably worked just fine because things that use `options1`/`options2` didn't differentiate between `in` and `undefined`? Or maybe it was just plain broken! Either way, I think this new code is less confusing.